packages/shared/lib/utils/utils.ts

```diff
@@ -117,13 +117,14 @@ export function getRedisUrl() {
 export function isValidHttpUrl(str: string) {
     const pattern = new RegExp(
         '^(https?:\\/\\/)?' + // protocol
-            '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|localhost|' + // domain name
+            '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|' + // domain name
+            'localhost|' + // OR localhost
             '((\\d{1,3}\\.){3}\\d{1,3}))' + // OR ip (v4) address
-            '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*' + // port and path
-            '(\\?[;&a-z\\d%_.~+=-]*)?' + // query string
-            '(\\#[-a-z\\d_]*)?$',
+            '(\\:\\d+)?(\\/[-a-z\\d%_.~+@]*)*' + // port and path
+            '(\\?[;&a-z\\d%_.~+=-@,]*)?' + // query string, including commas
+            '(\\#[-a-z\\d_]*)?$', // fragment locator
         'i'
-    ); // fragment locator
+    );
```

Comment:
can we just use the `new URL()` instead of a regex?

Reply:
Yeah, I suppose so. I'm unsure why the regex was used in the first place tbh but much easier to just use URL